lib/src/main/kotlin/com/swmansion/starknet/data/types/FeltArray.kt

```diff
@@ -1,11 +1,30 @@
 package com.swmansion.starknet.data.types
 
 import com.swmansion.starknet.data.types.conversions.ConvertibleToCalldata
+import com.swmansion.starknet.extensions.toFelt
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.builtins.ListSerializer
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
 
-data class FeltArray(private val list: MutableList<Felt>) : ConvertibleToCalldata, MutableList<Felt> by list {
+data class FeltArray(private val list: MutableList<Felt>) : ConvertibleToCalldata, MutableList<Felt> by list, HttpBatchRequestType {
     constructor(vararg elements: Felt) : this(elements.toMutableList())
     constructor(collection: Collection<Felt>) : this(collection.toMutableList())
     constructor() : this(emptyList())
 
-    override fun toCalldata(): List<Felt> = list.toList()
+    override fun toCalldata(): List<Felt> = listOf(size.toFelt) + list
+}
+
+object FeltArraySerializer : KSerializer<FeltArray> {
```

Comment:
also not sure about FeltArray, the name may be a bit vague. Maybe FeltList would be better?

Reply:
`List<Felt>` was returned in some methods of `Provider` (eg. [here](https://github.com/software-mansion/starknet-jvm/blob/895dfc81eb51d0e15d8fa72c12f04596c8eb106c/lib/src/main/kotlin/com/swmansion/starknet/provider/Provider.kt#L157)) and I had to change it to a wrapper class. `FeltArray` has been already implemented before, so I decided it can be utilized and there's no need to create another wrapper. `FeltArray` didn't have serializer, that's why I've added it. Should I add separate wrapper, eg. `FeltList` and use it instead of existing `FeltArray`, wdyt @THenry14 ?

Reply:
ok, we can leave it as is then I suppose